book/src/proposals/tick_verification.md

```diff
@@ -0,0 +1,65 @@
+# Tick Verification
+
+This design the criteria and validation of ticks in a slot. It also describes
+error handling and slashing conditions encompassing how the system handles
+transmissions that do not meet these requirements.
+
+# Slot structure
+
+Each slot must contain an expected `ticks_per_slot` number of ticks. The last
+shred in a slot must contain only the entirety of the last tick, and nothing 
+else. The leader must also mark this shred containing the last tick with the 
+`LAST_SHRED_IN_SLOT` flag. Between ticks, there must be `hashes_per_tick`
+number of hashes.
+
+# Handling bad transmissions
+
+Malicious transmissions `T` are handled in two ways:
+
+1) If a leader can generate some erronenous transmission `T` and also some 
+alternate transmission `T'` for the same slot without violating any slashing
+rules for duplicate transmissions (for instance if `T'` is a subset of `T`),
+then the cluster must handle the possibility of both transmissions being live.
+
+Thus this means we cannot mark the erronenous transmission `T` as dead because
+the cluster may have reached consensus on `T`. These cases necessitate a 
+slashing proof to punish this bad behavior.
+
+2) Otherwise, we can simply mark the slot as dead and not playable. A slashing
+proof may or may not be necessary depending on feasibility.
+
+# Blocktree receiving shreds
+
+When blocktree receives a new shred `s`, there are two cases:
+
+1) `s` is marked as `LAST_SHRED_IN_SLOT`, then check if there exists a shred 
+`s'` in blocktree for that slot where `s'.index > s.index` If so, together `s`
+and `s'` constitute a slashing proof.
+
+2) Blocktree has already received a shred `s'` marked as `LAST_SHRED_IN_SLOT`
+with index `i`. If `s.index > i`, then together `s` and `s'`constitute a 
+slashing proof. In this case, blocktree will also not insert `s`.
+
+
+# Replaying and validating ticks
```

Comment:
@jstarry I think these are the conditions related to your PR. 

The only case where we don't mark slots as dead is when we detect extra shreds that come after the `last tick` or `the last shred` (they should be one and the same under this proposal, and are handled by the above secction `Blocktree receiving shreds`)